Shared/mods/deathmatch/logic/luadefs/CLuaPathDefs.cpp

```diff
@@ -0,0 +1,93 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        Shared/mods/deathmatch/logic/luadefs/CLuaFileDefs.cpp
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+
+#ifndef MTA_CLIENT
+    // NOTE: Must be included before ILuaModuleManager.h which defines its own CChecksum type.
+    #include "CChecksum.h"
+#endif
+
+#include "CLuaPathDefs.h"
+#include "CScriptFile.h"
+#include "CScriptArgReader.h"
+#include <lua/CLuaFunctionParser.h>
+
+void CLuaPathDefs::LoadFunctions()
+{
+    constexpr static const std::pair<const char*, lua_CFunction> functions[]{
+        {"pathListDir", ArgumentParser<pathListDir>},
+        {"pathIsFile", ArgumentParser<pathIsFile>},
+        {"pathIsDirectory", ArgumentParser<pathIsDirectory>},
+    };
+
+    // Add functions
+    for (const auto& [name, func] : functions)
+        CLuaCFunctions::AddFunction(name, func);
+}
+
+void CLuaPathDefs::AddClass(lua_State* luaVM)
+{
+    lua_newclass(luaVM);
+
+    lua_classfunction(luaVM, "listDir", "pathListDir");
+    lua_classfunction(luaVM, "isFile", "pathIsFile");
+    lua_classfunction(luaVM, "isDirectory", "pathIsDirectory");
+
+    lua_registerclass(luaVM, "path");
+}
+
+std::optional<std::vector<std::string>> CLuaPathDefs::pathListDir(lua_State* luaVM, std::string path)
+{
+    CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+    if (!pLuaMain)
+        return std::nullopt;
+
+    SString strAbsPath;
+
+    CResource* pResource = pLuaMain->GetResource();
+    if (!CResourceManager::ParseResourcePathInput(path, pResource, &strAbsPath))
+        return std::nullopt;
+
+    if (!DirectoryExists(strAbsPath))
+        return std::nullopt;
+
```

Comment:
Warnings will be useful here

Reply:
What kind of warning here?

Reply:
"Unable to parse the path."
"Directory does not exist."
These warnings will be helpful during debugging and show potential issues.

Reply:
> "Unable to parse the path." "Directory does not exist." These warnings will be helpful during debugging and show potential issues.

sorry, but I've only done simple things with the parser. How would I throw a warning in ArgumentParser?

Reply:
Did you try `m_pScriptDebugging->LogWarning(luaVM, "message");` ?

Reply:
> Did you try `m_pScriptDebugging->LogWarning(luaVM, "message");` ?

`luaVM` is not accessible via ArgumentParser

Reply:
```
CLuaPathDefs::pathListDir(lua_State* luaVM, std::string path)
                                     ^^^^^^
 ```

Reply:
> ```
> CLuaPathDefs::pathListDir(lua_State* luaVM, std::string path)
>                                      ^^^^^^
> ```

oops, sorry my bad. I think Im too tired with it all 😄